website/docs/r/search_deployment.html.markdown

```diff
@@ -1,57 +1,99 @@
+  project_id   = mongodbatlas_project.example.id
+  cluster_name = mongodbatlas_advanced_cluster.example.name
   specs = [
     {
       instance_size = "S20_HIGHCPU_NVME"
-      node_count = 2
+      node_count    = 2
     }
   ]
 }
+
+data "mongodbatlas_search_deployment" "example" {
+  project_id   = mongodbatlas_search_deployment.example.project_id
+  cluster_name = mongodbatlas_search_deployment.example.cluster_name
+}
+
+output "mongodbatlas_search_deployment_id" {
+  value = data.mongodbatlas_search_deployment.example.id
+}
 ```
 
-## Argument Reference
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `cluster_name` (String) Label that identifies the cluster to return the search nodes for.
+- `project_id` (String) Unique 24-hexadecimal character string that identifies the project.
+- `specs` (Attributes List) List of settings that configure the search nodes for your cluster. (see [below for nested schema](#nestedatt--specs))
+
+### Optional
+
+- `timeouts` (Attributes) (see [below for nested schema](#nestedatt--timeouts))
+
+### Read-Only
 
-* `project_id` - (Required) Unique 24-hexadecimal digit string that identifies your project.
-* `cluster_name` - (Required) Label that identifies the cluster to create search nodes for.
-* `specs` - (Required) List of settings that configure the search nodes for your cluster. This list is currently limited to defining a single element. See [specs](#specs).
-* `timeouts`- (Optional) The time to wait for search nodes to be created, updated, or deleted. The timeout value is defined by a signed sequence of decimal numbers with a time unit suffix such as: `1h45m`, `300s`, `10m`, .... The valid time units are:  `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`. The attribute must be defined with [nested attributes](https://developer.hashicorp.com/terraform/plugin/framework/resources/timeouts#attribute). The default timeout for create, update, and delete is `3h`. Learn more about timeouts [here](https://developer.hashicorp.com/terraform/plugin/framework/resources/timeouts).
+- `id` (String) Unique 24-hexadecimal digit string that identifies the search deployment.
+- `state_name` (String) Human-readable label that indicates the current operating condition of this search deployment.
 
-### Specs
+<a id="nestedatt--specs"></a>
+### Nested Schema for `specs`
 
-Specs list is defined as a ["list nested attribute"](https://developer.hashicorp.com/terraform/plugin/framework/handling-data/attributes/list-nested) containing a single element.
+Required:
 
-* `instance_size` - (Required) Hardware specification for the search node instance sizes. The [MongoDB Atlas API](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Atlas-Search/operation/createAtlasSearchDeployment) describes the valid values. More details can also be found in the [Search Node Documentation](https://www.mongodb.com/docs/atlas/cluster-config/multi-cloud-distribution/#search-tier).
```

Comment:
Interesting: are we losing some valuable information by removing this?

Reply:
If the description is not good enough, we should update the open API spec of the endpoint. In this case, I will open a follow-up PR to add "More details can also be found in the [Search Node Documentation](https://www.mongodb.com/docs/atlas/cluster-config/multi-cloud-distribution/#search-tier)" to the endpoint. Thanks!

Reply:
FYI created this ticket https://jira.mongodb.org/browse/CLOUDP-222701 to update the open api spec